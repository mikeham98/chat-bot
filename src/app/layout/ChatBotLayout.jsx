import React, {Suspense} from 'react';
import messagesReducer from "../reducers/messages/index.reducers";
import store from "../store";
import styles from '../../themes/components/layout.scss';

const ChatBotContainer = React.lazy(() =>
    import(/* webpackChunkName: "MESSAGES_LIST_CONTAINER" */ "../containers/ChatBot/index")
        .then(() => {
            store.injectReducer('messages', messagesReducer);
            return import ("../containers/ChatBot/index");
        })
);

export default class ChatBotLayout extends React.Component {
    render() {
        return (
            <Suspense fallback={<div>Loading...</div>}>
                    <ChatBotContainer/>
            </Suspense>
        )
    }
}