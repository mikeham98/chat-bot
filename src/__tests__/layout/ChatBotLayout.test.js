import React from 'react';
import {shallow} from 'enzyme';
import ChatBotLayout from '../../app/layout/ChatBotLayout';

describe('ChatBotLayout', () => {

    let shallowedWrapper;
    let props = {};
    const wrapper = () => {
        if (!shallowedWrapper) {
            shallowedWrapper = shallow(<ChatBotLayout {...props}/>);
        }
        return shallowedWrapper;
    };

    beforeEach(() => {
        props = {};
        shallowedWrapper = undefined;
    });

    describe('render', () => {
        it('should return 2 lazy components (HeaderContainer and ChatBotContainer)', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });
});