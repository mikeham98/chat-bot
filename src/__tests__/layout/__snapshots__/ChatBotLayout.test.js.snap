// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatBotLayout render should return 2 lazy components (HeaderContainer and ChatBotContainer) 1`] = `
<Suspense
  fallback={
    <div
      className="centerChatBotSpinner"
    >
      <Memo(Spinner) />
    </div>
  }
>
  <div
    className="headerLayout"
  >
    <lazy />
  </div>
  <lazy />
</Suspense>
`;
